ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/date/DateTaskFragment.kt

```diff
@@ -40,6 +42,10 @@ class DateTaskFragment : AbstractTaskFragment<DateTaskViewModel>() {
     taskBinding.lifecycleOwner = this
     taskBinding.fragment = this
     taskBinding.viewModel = viewModel
+    if (viewModel.taskTaskData.value.isNotNullOrEmpty()) {
+      val timestamp = (viewModel.taskTaskData.value as DateTimeTaskData).timeInMillis
+      viewModel.updateResponse(getDateFormatter(), Date(timestamp))
```

Comment:
We already update the response with initial value when loading from draft. So, here we are doing this for updating the UI and not the response. The issue is that text displayed on the UI is tied to `updateResponse` of `DateTaskViewModel`. Can we pull the `dateText` state flow from view model to the fragment to fix this dependency?

Reply:
Keeping it in ViewModel helps in managing the state of the dateText, even in config change, is shifting it to fragment will be a good idea?

Reply:
You can chain the `dateText` state flow with view model's task data. Then everything will work as expected even for config changes.